hyundai_kia_connect_api/HyundaiBlueLinkAPIUSA.py

```diff
@@ -117,47 +126,79 @@ def login(self, username: str, password: str) -> Token:
-        headers = self.API_HEADERS
-        headers["accessToken"] = token.access_token
-        headers["vin"] = vehicle.VIN
-        headers["username"] = token.username
-        headers["blueLinkServicePin"] = token.pin
+        headers = self._get_vehicle_headers(token, vehicle)
+        if refresh:
+            headers["REFRESH"] = "true"
 
-        _LOGGER.debug(f"{DOMAIN} - using API headers: {self.API_HEADERS}")
+        _LOGGER.debug(f"{DOMAIN} - using API headers: {headers}")
 
         response = self.sessions.get(url, headers=headers)
         response = response.json()
-        _LOGGER.debug(f"{DOMAIN} - get_cached_vehicle_status response {response}")
+        _LOGGER.debug(f"{DOMAIN} - get_vehicle_status response {response}")
 
-        vehicle_status = {}
-        vehicle_status["vehicleStatus"] = response["vehicleStatus"]
+        status = dict(response["vehicleStatus"])
 
-        vehicle_status["vehicleStatus"]["dateTime"] = (
-            vehicle_status["vehicleStatus"]["dateTime"]
+        status["dateTime"] = (
+            status["dateTime"]
             .replace("-", "")
             .replace("T", "")
             .replace(":", "")
             .replace("Z", "")
         )
 
-        vehicle_status["vehicleDetails"] = self._get_vehicle(token, vehicle)
+        return status
 
-        if vehicle.odometer:
-            if vehicle.odometer < get_child_value(
-                vehicle_status["vehicleDetails"], "odometer"
-            ):
-                vehicle_status["vehicleLocation"] = self.get_location(token, vehicle)
+    def _get_vehicle_location(self, token: Token, vehicle: Vehicle):
+        r"""
+        Get the location of the vehicle
+        This logic only checks odometer move in the update.  This call doesn't protect from overlimit as per:
+        Only update the location if the odometer moved AND if the last location update was over an hour ago.
+        Note that the "last updated" time is initially set to three hours ago.
+        This will help to prevent too many calls to the API
+        """
+        url = self.API_URL + "rcs/rfc/findMyCar"
+        headers = self._get_vehicle_headers(token, vehicle)
+        try:
+            response = self.sessions.get(url, headers=headers)
+            response_json = response.json()
+            _LOGGER.debug(f"{DOMAIN} - Get Vehicle Location {response_json}")
+            if response_json.get("coord") is not None:
+                return response_json
             else:
-                vehicle_status["vehicleLocation"] = None
-        else:
-            vehicle_status["vehicleLocation"] = self.get_location(token, vehicle)
-        return vehicle_status
+                if (
+                    response_json.get("errorCode", 0) == 502
+                    and response_json.get("errorSubCode", "") == "HT_534"
+                ):
+                    _LOGGER.warn(
+                        f"{DOMAIN} - get vehicle location rate limit exceeded."
```

Comment:
Longer term we have exceptions this should throw that are defined as standard.  Since we don't have a dev with cars in this region it lags behind.